backend/server/src/client.rs

```diff
@@ -29,7 +41,25 @@ impl UnixService {
     }
 
     pub async fn run(self) {
-        listen_for_client(self.listener).await
+        self.listen_for_client().await
+    }
+
+    async fn listen_for_client(self) {
+        loop {
+            match self.listener.accept().await {
+                Ok((stream, _)) => {
+                    let new_dispatch = self.dispatch.clone();
+                    tokio::spawn(async {
+                        if let Err(err) = handle_client(stream, new_dispatch).await {
+                            warn!("Failed to handle socket connection: {:?}", err);
```

Comment:
we could also make `handle_client` a method of `DispatchChannels`, then this would just be:
```
dispatch.clone().handle_client(stream);
``

Reply:
Yeah, but on the other hand, it is not really the dispatch's "job" to handle the client. Thinking into the future, the dispatch will be something non-trivial that can handle multi-machine communication and stuff (per our previous discussions on Discord), so I would not overload it with this functionality as well.

Reply:
very true.... this is probably me just early optimizing.